src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs

```diff
@@ -749,6 +760,9 @@ internal SaveChangesData PrepareForSaveChanges()
 
             PrepareCompareExchangeEntities(result);
 
+            RemoveCountersFromCache(CountersToIncrement);
```

Comment:
Why are you removing it? The result of the Inc operation will already tell you what the new value is, no?

Reply:
fixed, now we use the inc op result to update counters cache